gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageReadChannel.java

```diff
@@ -215,39 +215,39 @@ protected GoogleCloudStorageItemInfo getInitialMetadata() throws IOException {
     return null;
   }
 
-  /** Returns {@link GoogleCloudStorageItemInfo} used to initialize metadata in constructor. */
-  private GoogleCloudStorageItemInfo fetchInitialMetadata() throws IOException {
-    StorageObject object;
+  /** Method used to cache footer and initializes metadata in constructor. */
+  private void prefetchFooterAndInitMetadata() throws IOException {
+    checkState(!metadataInitialized, "metadata should not be initialized yet for '%s'", resourceId);
+
+    // TODO: use metadata request if `readOptions.getMinRangeRequestSize()` is 0.
```

Comment:
what is the point making this call when MinRangeRequestSize is 0. i.e. what will happen with this change if MinRangeRequestSize is 0?